# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PContent.ipdl
# Commit: 604c4c52574a
# Full Hash: 604c4c52574ae48f4f168eb999bc0af15dd5e0d6
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-04-15 21:21:06
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P2 PRemoteWorkerDebuggerMangaer binding implementation. r=asuth
#   
#   This patch implements the PRemoteWorkerDebuggerManager binding.
#   
#   PRemoteWorkerDebuggerManager is the IPC that handles the registration/unregistration of remote debuggers.
# ==============================================================================

diff -r 77eda17d0795 -r 604c4c52574a dom/ipc/PContent.ipdl
--- a/dom/ipc/PContent.ipdl	Tue Apr 15 09:47:38 2025 +0000
+++ b/dom/ipc/PContent.ipdl	Tue Apr 15 09:47:39 2025 +0000
@@ -42,6 +42,7 @@
 include protocol PURLClassifierLocalByName;
 include protocol PVRManager;
 include protocol PRemoteDecoderManager;
+include protocol PRemoteWorkerDebuggerManager;
 include protocol PRemoteWorkerService;
 include protocol PProfiler;
 include protocol PScriptCache;
@@ -784,7 +785,8 @@
     /**
      * Initialize the RemoteWorkerService thread in the content process.
      */
-    async InitRemoteWorkerService(Endpoint<PRemoteWorkerServiceChild> aEndpoint);
+    async InitRemoteWorkerService(Endpoint<PRemoteWorkerServiceChild> aEndpoint,
+        Endpoint<PRemoteWorkerDebuggerManagerChild> aDebuggerChildEp);
 
     /**
      * Send BlobURLRegistrationData to child process.